examples/direct_fidelity_estimation.py

```diff
@@ -207,20 +207,29 @@ def direct_fidelity_estimation(circuit: cirq.Circuit, qubits: List[cirq.Qid],
     Args:
         circuit: The circuit to run the simulation on.
         qubits: The list of qubits.
-        noise: The noise model when doing a simulation.
+        sampler: Either a noisy simulator or an engine.
         n_trial: The total number of Pauli measurements.
         n_clifford_trials: In case the circuit is Clifford, we specify the
             number of trials to estimate the noise-free pauli traces.
         samples_per_term: is set to 0, we use the 'noise' parameter above and
             simulate noise in the circuit. If greater than 0, we ignore the
             'noise' parameter above and instead run an estimation of the
             characteristic function.
+        log_output_filename: If not empty, the filename where we write the log
+            of the activity. Useful when running experiments and we want to
+            store data.
 
     Returns:
         The estimated fidelity.
     """
     # n_trials is upper-case N in https://arxiv.org/abs/1104.3835
 
+    if log_output_filename:
+        sys.stdout = open(log_output_filename, "w")
```

Comment:
This breaks the [principle of separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns). Function for direct fidelity estimation should not be responsible for setting up logging.

Perhaps we don't really need to print anything in this function. It seems that what we print are the inputs and return value anyway.

Reply:
I had a concern that it might not be the right way. I reverted the changes to the logging, and instead narrowed the scope to only the refactoring.